### PR TITLE
Fixed show-call-sequences feature of goto instrument; added test suite

### PR DESCRIPTION
**Pull request by KPouliasis**
The current version of cbmc will fail in the last set of this test-suite (show-call-sequences3)

### CODE REVIEW

File: regression/goto-instrument/show-call-sequences2/test.desc [hunk 1]
```diff
@@ -0,0 +1,9 @@
+CORE
+main.c
+--show-call-sequences
+^EXIT=0$
+^SIGNAL=0$
+^VERIFICATION SUCCESSFUL$
+main -> foo
+foo -> foo
+--
```
Comment (peterschrammel) [inline regression/goto-instrument/show-call-sequences2/test.desc:9]: Add the unexpected cases

File: regression/goto-instrument/show-call-sequences3/main.c [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+void recurse(int low) {
+   if(low >= 2) {
+      recurse(low-1);
+      recurse(low-2);
+   } else {
+      foo(2);
+      foo(3);
+      return;
+   }
+}
```
Comment (peterschrammel) [inline regression/goto-instrument/show-call-sequences3/main.c]: add a newline at end of file

File: regression/goto-instrument/show-call-sequences1/main.c [hunk 3]
```diff
@@ -0,0 +1,12 @@
+int foo(int x){
```
Comment (peterschrammel) [inline regression/goto-instrument/show-call-sequences1/main.c:1]: ideally, test cases should follow coding guidelines too

File: regression/goto-instrument/show-call-sequences3/test.desc [hunk 4]
```diff
@@ -0,0 +1,9 @@
+CORE
+main.c
+--show-call-sequences
+^SIGNAL=0$
+recurse -> recurse
```
Comment (peterschrammel) [inline regression/goto-instrument/show-call-sequences3/test.desc]: check EXIT

File: src/goto-instrument/call_sequences.cpp [hunk 5]
```diff
@@ -21,14 +21,15 @@ Date: April 2013
...
-  const goto_programt &goto_program,
-  const goto_programt::const_targett start)
+  const irep_idt &caller,
+  const goto_programt &goto_program)
 {
-  std::cout << "# From " << function << '\n';
-
+  // show calls in  blocks within caller body
+  // dfs on code blocks using stack
+  std::cout << "# " << caller << '\n';
```
Comment (peterschrammel) [inline src/goto-instrument/call_sequences.cpp:31]: Is this output desired? If so, then it should be tested.

Comment (KPouliasis) [reply src/goto-instrument/call_sequences.cpp:31]: addressed in latest commit by adding new test

File: regression/goto-instrument/show-call-sequences1/main.c (hunk 3 above)
Comment (KPouliasis) [reply regression/goto-instrument/show-call-sequences1/main.c:1]: addressed in latest commit

File: regression/goto-instrument/show-call-sequences2/test.desc (hunk 1 above)
Comment (KPouliasis) [reply regression/goto-instrument/show-call-sequences2/test.desc:9]: addressed  in latest commit

File: regression/goto-instrument/show-call-sequences2/test.desc [hunk 6]
```diff
@@ -7,3 +7,4 @@ main.c
 main -> foo
 foo -> foo
 --
+foo -> main
```
Comment (peterschrammel) [inline regression/goto-instrument/show-call-sequences2/test.desc]: newline at the end of file

Comment (peterschrammel) [reply regression/goto-instrument/show-call-sequences2/test.desc]: what about `main -> main`?

File: regression/goto-instrument/show-call-sequences2/main.c [hunk 7]
```diff
@@ -1,12 +1,17 @@
...
+  else
+  {
     return foo(3);
   }
 }
 
-int main() {
+int main()
+{
   foo(0);
```
Comment (peterschrammel) [inline regression/goto-instrument/show-call-sequences2/main.c]: return ?

File: regression/goto-instrument/show-call-sequences2/main.c [hunk 8]
```diff
@@ -1,12 +1,17 @@
...
 }
+
```
Comment (peterschrammel) [inline regression/goto-instrument/show-call-sequences2/main.c]: remove blank line

File: regression/goto-instrument/show-call-sequences3/main.c [hunk 9]
```diff
@@ -1,19 +1,31 @@
...
+    foo(2);
+    foo(3);
+    return;
+  }
+}
+
+int main()
+{
+  recurse(5);
+}
```
Comment (peterschrammel) [inline regression/goto-instrument/show-call-sequences3/main.c]: return?

File: regression/goto-instrument/show-call-sequences4/main.c [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+{
+  return; 
+}
+
+void boo()
+{
+  return;
+}
+
+
```
Comment (peterschrammel) [inline regression/goto-instrument/show-call-sequences4/main.c]: remove two blank lines

File: regression/goto-instrument/show-call-sequences4/main.c [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+
+int main()
+{
+  moo();
+  foo();
+}
```
Comment (peterschrammel) [inline regression/goto-instrument/show-call-sequences4/main.c]: return?

File: regression/goto-instrument/show-call-sequences4/test.desc [hunk 12]
```diff
@@ -0,0 +1,9 @@
+CORE
+main.c
+--show-call-sequences
+activate-multi-line-match
+main -> moo\nmain -> foo
+foo -> moo\nfoo -> boo
+EXIT=0
+SIGNAL=0
+
```
Comment (peterschrammel) [inline regression/goto-instrument/show-call-sequences4/test.desc]: add unexpected cases

Comment (KPouliasis) [reply regression/goto-instrument/show-call-sequences4/test.desc]: This is supposed to show case that function calls per function are grouped together - hence the multiline. Of course, unexpected cases wont' hurt. But that's the feature that the test is supposed to verify

**Comment by peterschrammel** [comment]
@KPouliasis, please rebase.

**Comment by KPouliasis** [comment]
Squashed all my commits to a meaningful one. Michael kindly rebased on :develop.
@peterschrammel

File: regression/goto-instrument/show-call-sequences2/main.c [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+  {
+    return foo(3);
+  }
+}
+
+int main()
+{
+  foo(0);
+  return 0;
+}
```
Comment (peterschrammel) [inline regression/goto-instrument/show-call-sequences2/main.c:17]: Newline missing

File: regression/goto-instrument/show-call-sequences4/main.c [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+{
+  return;
+}
+
+int main()
+{
+  moo();
+  foo();
+  return 0;
+}
```
Comment (peterschrammel) [inline regression/goto-instrument/show-call-sequences4/main.c:22]: Newline missing

File: src/goto-instrument/call_sequences.cpp [hunk 15]
```diff
@@ -59,52 +57,16 @@ void show_call_sequences(
...
 }
-
 void show_call_sequences(const goto_functionst &goto_functions)
 {
   // do per function
-
+  // show the calls in the body of the specific function
   forall_goto_functions(f_it, goto_functions)
     show_call_sequences(f_it->first, f_it->second.body);
-}
```
Comment (peterschrammel) [inline src/goto-instrument/call_sequences.cpp]: Moving these braces seems random

**Pull request merged**